### PR TITLE
Move Hangul Syllable composition and decomposition to [unic-ucd-hangul]

### PR DESCRIPTION
**Pull request by eyeplum**
Implement #209

### CODE REVIEW

File: unic/ucd/hangul/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "unic-ucd-hangul"
+version = "0.7.0"
+authors = ["The UNIC Project Developers"]
+repository = "https://github.com/behnam/rust-unic/"
+license = "MIT/Apache-2.0"
+description = "UNIC — Unicode Character Database — Hangul"
```
Comment (behnam) [inline unic/ucd/hangul/Cargo.toml]: Could you please elaborate on what the create is here?

Maybe "Hangul Syllable Composition & Decomposition", to follow the name of the algorithms in TUS (The Unicode Standard)?

File: unic/ucd/hangul/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// except according to those terms.
+
+#![no_std]
+#![forbid(future_incompatible, missing_debug_implementations, unconditional_recursion, unsafe_code)]
+#![deny(bad_style, unsafe_code, unused)]
+
+extern crate unic_ucd_version;
+use unic_ucd_version::UnicodeVersion;
+
+pub mod hangul;
```
Comment (behnam) [inline unic/ucd/hangul/src/lib.rs]: Since `hangul` algorithms are all this component providing, let's make the module here private and expose the API via `pub use`.

Also, what do you think about renaming the public API functions to `compose_syllable()` and `decompose_syllable()`.

IMHO, it would reflect the algorithm better, and is more readable.

File: unic/ucd/hangul/tests/basic_tests.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        let mut collect_decomposed = |chr| {
+            decomposed_lvt.push(chr);
+        };
+        hangul::decompose('퓛', &mut collect_decomposed);
+    }
+    assert_eq!(decomposed_lvt, ['ᄑ', 'ᅱ', 'ᆶ']);
+}
+
+#[test]
+fn test_compose() {
```
Comment (behnam) [inline unic/ucd/hangul/tests/basic_tests.rs]: I missed this test file before. Looks good. Maybe we can add more function to it that decomposes and the composes, and use that in the docblock tests?

(Or you can leave the test additions fro another PR. Up to you.)

File: unic/ucd/hangul/Cargo.toml (hunk 1 above)
Comment (eyeplum) [reply unic/ucd/hangul/Cargo.toml]: Fixed :)

File: unic/ucd/hangul/src/lib.rs (hunk 2 above)
Comment (eyeplum) [reply unic/ucd/hangul/src/lib.rs]: Agreed, refined the APIs.

File: unic/ucd/hangul/tests/basic_tests.rs (hunk 3 above)
Comment (eyeplum) [reply unic/ucd/hangul/tests/basic_tests.rs]: Added a new test case `test_decompose_then_compose` for this.
Also used the new test case in the docblock tests.

**Comment by eyeplum** [comment]
Thanks for the review @behnam!
I have updated the PR according to the suggestions.
Also added a simple `README.md` file.

**Comment by behnam** [comment]
bors: r+
